### PR TITLE
Data model intro paragraphs

### PR DESCRIPTION
**Pull request by Dhghomon**
Feel free to take an axe to anything here

### CODE REVIEW

**Comment by 1st1** [comment]
I like it! Dave, can you reformat it to be 79 characters max per line? Otherwise our linter would not allow us to merge this.

**Comment by 1st1** [comment]
This is for https://edgedb-staging.vercel.app/docs/datamodel/index

File: docs/datamodel/index.rst [hunk 1]
```diff
@@ -4,7 +4,13 @@
 Data Model
 ==========
 
-EdgeDB is a relational database with strongly typed schema.
+On this page you'll find an overview of the fundamental types, objects, and concepts for EdgeDB, a relational database with strongly typed schema. Here's a quick overview of the links below:
```
Comment (1st1) [inline docs/datamodel/index.rst:7]: "On this page" -> "In this section"?

File: docs/datamodel/index.rst [hunk 2]
```diff
@@ -4,7 +4,13 @@
...
+
+Types in EdgeDB include your own **Object Types** (e.g. *User*) and **Abstract Types** for other types to extend (e.g. *HasEmailAddress* for *User* and others can inherit), plus **Scalar Types** with single values (str, int64, etc.) and **Collection Types** like **arrays** and **tuples** for multiple values.
```
Comment (1st1) [inline docs/datamodel/index.rst]: Perhaps we should hyperlink `**Object Types**`, `**Abstract Types**`, and all other terms to their respective pages. I can quickly do that myself if you don't feel comfortable with ReST.

Comment (Dhghomon) [reply docs/datamodel/index.rst]: Ah, it's ReST? I tried using just Markdown and it wasn't doing any hyperlinking so resorted to bolding in the meantime. Thanks!

**Review by elprans — APPROVED** [review]

File: docs/datamodel/index.rst [hunk 3]
```diff
@@ -4,7 +4,28 @@
...
+notes for others), **constraints** to set limits (e.g. maximum length, 
+minimum value, or even create your own), **indexes** for faster querying, 
+and **computables** to use expressions to define properties or links 
+(e.g. *property email := .user_name ++ '@' ++ .provider_name*).
+
+**Expression Aliases** let you use existing types under new names to build
+on them without touching the original - both in your schema or on the fly 
+inside a query. You can also create your own **functions**, strongly typed 
+along with everything else in EdgeDB. All this goes into the schema under 
+one or more **modules** (namespaces).
```
Comment (1st1) [inline docs/datamodel/index.rst:28]:
```suggestion
In this section you'll find an overview of the fundamental types, objects,
and concepts for EdgeDB, a relational database with strongly typed schema.
Here's a quick overview of the links below:

Types in EdgeDB include your own **Object Types** (e.g. *User*) and
**Abstract Types** for other types to extend (e.g. *HasEmailAddress* for
*User* and others can inherit), plus **Scalar Types** with single values
(str, int64, etc.) and **Collection Types** like **arrays** and **tuples**
for multiple values.

Start putting your **Object Types** together with **properties** and
**links**. Then build on them with items like **annotations** (readable
notes for others), **constraints** to set limits (e.g. maximum length,
minimum value, or even create your own), **indexes** for faster querying,
and **computables** to use expressions to define properties or links
(e.g. *property email := .user_name ++ '@' ++ .provider_name*).

**Expression Aliases** let you use existing types under new names to build
on them without touching the original - both in your schema or on the fly
inside a query. You can also create your own **functions**, strongly typed
along with everything else in EdgeDB. All this goes into the schema under
one or more **modules** (namespaces).
```

**Review by 1st1 — APPROVED** [review]

**Comment by 1st1** [comment]
Thanks Dave!

**Pull request merged**